### ISSUE TITLE
Conditional based on a JSON object within an each loop

### ISSUE THREAD

**Issue by christinalau** [jadejs/jade#2011]
I am getting an error when trying to do a if test of a json object like this:

```js
tr(ng-repeat='item in deviceVariable | offset: currentPage*itemsPerPage | limitTo: itemsPerPage')
          td {{item.devicename}} 
          td {{item.gatewayip}}
          td {{item.deviceid}}
          if #{item.status} == "success"
            td.success {{item.last_read_date}} 
          else
            td.danger {{item.last_read_date}}
          td
```

This is the error I am getting. Can you help me fix this or is this a bug? I don't seem to be able to google a soln here. Thank you...

```
    39|           td {{item.gatewayip}}
    40|           td {{item.deviceid}}
  > 41|           if #{item.status} == "success"
    42|             td.success {{item.last_read_date}} 
    43|           else
    44|             td.danger {{item.last_read_date}}

Unexpected token ILLEGAL
    at Function (native)
    at assertExpression (/usr/local/lib/node_modules/jade/lib/lexer.js:30:33)
    at Object.Lexer.conditional (/usr/local/lib/node_modules/jade/lib/lexer.js:509:11)
    at Object.Lexer.next (/usr/local/lib/node_modules/jade/lib/lexer.js:914:15)
    at Object.Lexer.lookahead (/usr/local/lib/node_modules/jade/lib/lexer.js:113:46)
    at Parser.lookahead (/usr/local/lib/node_modules/jade/lib/parser.js:102:23)
    at Parser.peek (/usr/local/lib/node_modules/jade/lib/parser.js:79:17)
    at Parser.tag (/usr/local/lib/node_modules/jade/lib/parser.js:811:30)
    at Parser.parseTag (/usr/local/lib/node_modules/jade/lib/parser.js:738:17)
    at Parser.parseExpr (/usr/local/lib/node_modules/jade/lib/parser.js:211:21)
```

**Comment by jadaradix** [comment]
TLDR: Use interpolation for mark up, access objects directly in logic (<code>if</code> is logic; it is not a tag).

Here's your fix:

    if item.status == "success"

It's different to a <code>td</code> because a <code>td</code> is a HTML tag, so interpolation works. But a jade <code>if</code> isn't a tag; it's a "logical statement" (or whatever they are called these days). JavaScript is parsing <code>#{item.status}</code> which makes no sense; <code>item</code> is a JavaScript object coming from the renderer so you should use it directly.

I think this issue can be closed. ^

**Comment by TimothyGu** [comment]
@jadaradix pretty much covered everything. Closing.

**Comment by ForbesLindesay** [comment]
I think the actual issue here was probably mixing jade and angular.  If you want a block of code to be conditional on angular values, you must use whatever the angular syntax for conditionals is.  If you want to use jade locals, you can use jade syntax for the conditional, but you should then use jade syntax throughout.
